src/v/datalake/schema_avro.cc

```diff
@@ -41,7 +38,7 @@ struct_from_avro_record(const avro::NodePtr& node, state& state) {
         auto field = std::move(result.value());
         if (field.has_value()) {
             ret.fields.push_back(iceberg::nested_field::create(
-              state.next_field_id(),
+              0,
```

Comment:
consider a named constant that is "placeholder_field_id" with a comment mentioning how it is assigned later

Reply:
good point, will do it